src/json-rpc-socket.ts

```diff
@@ -89,33 +89,61 @@ export class JsonRpcSocket {
    * Sends a JSON-RPC request through the socket and keeps a listener open to read associated responses as they arrive.
    * Returns a close method to clean up the listener.
    */
-  subscribe(request: JsonRpcRequest, listener: (response: JsonRpcResponse) => void): { close: () => void } {
-    request.id ??= uuidv4();
+  async subscribe(request: JsonRpcRequest, listener: (response: JsonRpcResponse) => void): Promise<{
+    response: JsonRpcResponse;
+    close?: () => Promise<void>;
+   }> {
+
+    if (!request.method.startsWith('rpc.subscribe.')) {
+      throw new Error('subscribe rpc requests must include the `rpc.subscribe` prefix');
+    }
+
+    if (!request.subscribe) {
+      throw new Error('subscribe rpc requests must include subscribe options');
+    }
+
+    const subscriptionId = request.subscribe.id;
     const messageHandler = (event: { data: any }):void => {
```

Comment:
consider renaming to `socketMessageEventListener` or `socketEventListener` 'coz:

1. I got it confused with DWN message handler...
2. It's consistent with `add/removeEvenListener()` usage.

Reply:
Done